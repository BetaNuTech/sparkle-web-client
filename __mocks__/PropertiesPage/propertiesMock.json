[
  {
    "id": 2,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2Fea35bb17a8b841e60c1d.jpeg?alt=media&token=2f85465d-3593-4c39-93ae-4165a2a1851e",
    "name": "A1400 Chestnut",
    "addr1": "1221 New Meister Lane",
    "addr2": "Blugerville, TX 78660",
    "city": "Zouisville",
    "state": "KY",
    "numOfDeficientItems": 82,
    "numOfFollowUpActionsForDeficientItems": 0,
    "numOfRequiredActionsForDeficientItems": 82,
    "lastInspectionDate": 1595501032,
    "lastInspectionScore": 92.72727272727272
  },
  {
    "id": 3,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2F3e1c1a56d1bd381af398.jpeg?alt=media&token=5d69f6db-f240-4db9-9e52-843c66ee1e05",
    "name": "CEmerson",
    "addr1": "1221 New Meister Lane",
    "addr2": "Pflugerville, TX 78660",
    "city": "Louisville",
    "state": "AY",
    "numOfDeficientItems": 0,
    "numOfFollowUpActionsForDeficientItems": 0,
    "numOfRequiredActionsForDeficientItems": 10,
    "lastInspectionDate": 1595521032,
    "lastInspectionScore": 92.72727272727272
  },
  {
    "id": 1,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2F3e1c1a56d1bd381af398.jpeg?alt=media&token=5d69f6db-f240-4db9-9e52-843c66ee1e05",
    "name": "BWalnut Ridge",
    "addr1": "1221 New Meister Lane",
    "addr2": "Pflugerville, TX 78660",
    "city": "Louisville",
    "state": "CY",
    "numOfDeficientItems": 11,
    "numOfFollowUpActionsForDeficientItems": 55,
    "numOfRequiredActionsForDeficientItems": 2,
    "lastInspectionDate": 1595621032,
    "lastInspectionScore": 92.72727272727272
  },
  {
    "id": 4,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2FdvSsHLv8cxAvIMKv9Gk0.jpeg?alt=media&token=f226942e-bc2e-476f-89dd-d2eee28cd2e4",
    "name": "EEmerson",
    "addr1": "1221 New Meister Lane",
    "addr2": "Pflugerville, TX 78660",
    "city": "Aruisville",
    "state": "KY",
    "numOfDeficientItems": 5,
    "numOfFollowUpActionsForDeficientItems": 24,
    "numOfRequiredActionsForDeficientItems": 1,
    "lastInspectionDate": 1595521032,
    "lastInspectionScore": 92.72727272727272
  },
  {
    "id": 5,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2FdvSsHLv8cxAvIMKv9Gk0.jpeg?alt=media&token=f226942e-bc2e-476f-89dd-d2eee28cd2e4",
    "name": "DWalnut Ridge",
    "addr1": "1221 New Meister Lane",
    "addr2": "Pflugerville, TX 78660",
    "city": "Louisville",
    "state": "BY",
    "numOfDeficientItems": 1,
    "numOfFollowUpActionsForDeficientItems": 22,
    "numOfRequiredActionsForDeficientItems": 22,
    "lastInspectionDate": 1593521032,
    "lastInspectionScore": 92.72727272727272
  }
]
