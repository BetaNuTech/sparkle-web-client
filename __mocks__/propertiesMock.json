[
  {
    "id": 2,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2Fea35bb17a8b841e60c1d.jpeg?alt=media&token=2f85465d-3593-4c39-93ae-4165a2a1851e",
    "name": "A1400 Chestnut",
    "addr1": "1221 New Meister Lane",
    "addr2": "Blugerville, TX 78660",
    "city": "Zouisville",
    "state": "KY",
    "lastInspectionDate": 1595501032,
    "lastInspectionScore": 92.72727272727272
  },
  {
    "id": 3,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2F3e1c1a56d1bd381af398.jpeg?alt=media&token=5d69f6db-f240-4db9-9e52-843c66ee1e05",
    "name": "CEmerson",
    "addr1": "1221 New Meister Lane",
    "addr2": "Pflugerville, TX 78660",
    "city": "Louisville",
    "state": "AY",
    "lastInspectionDate": 1595521032,
    "lastInspectionScore": 92.72727272727272
  },
  {
    "id": 1,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2F3e1c1a56d1bd381af398.jpeg?alt=media&token=5d69f6db-f240-4db9-9e52-843c66ee1e05",
    "name": "BWalnut Ridge",
    "addr1": "1221 New Meister Lane",
    "addr2": "Pflugerville, TX 78660",
    "city": "Louisville",
    "state": "CY",
    "lastInspectionDate": 1595621032,
    "lastInspectionScore": 92.72727272727272
  },
  {
    "id": 4,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2FdvSsHLv8cxAvIMKv9Gk0.jpeg?alt=media&token=f226942e-bc2e-476f-89dd-d2eee28cd2e4",
    "name": "EEmerson",
    "addr1": "1221 New Meister Lane",
    "addr2": "Pflugerville, TX 78660",
    "city": "Aruisville",
    "state": "KY",
    "lastInspectionDate": 1595521032,
    "lastInspectionScore": 92.72727272727272
  },
  {
    "id": 5,
    "backgroundImage": "https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2FdvSsHLv8cxAvIMKv9Gk0.jpeg?alt=media&token=f226942e-bc2e-476f-89dd-d2eee28cd2e4",
    "name": "DWalnut Ridge",
    "addr1": "1221 New Meister Lane",
    "addr2": "Pflugerville, TX 78660",
    "city": "Louisville",
    "state": "BY",
    "lastInspectionDate": 1593521032,
    "lastInspectionScore": 92.72727272727272
  }
]
