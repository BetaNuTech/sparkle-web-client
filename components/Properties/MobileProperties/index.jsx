import PropTypes from 'prop-types';
import styles from './MobileProperties.module.scss';
import { TeamItem } from './TeamItem';
import { PropertyItem } from './PropertyItem';

export const MobileProperties = ({ properties }) => (
  <ul className={styles.mobileProperties}>
    <li className={styles.mobileProperties__item}>
      <header>teams</header>
      <TeamItem name="Team one" />
      <TeamItem name="Team two" />
    </li>

    <li className={styles.mobileProperties__item}>
      <header>properties</header>

      {properties.map((property) => (
        <PropertyItem key={property.id} property={property} />
      ))}
    </li>
  </ul>
);

MobileProperties.propTypes = {
  properties: PropTypes.arrayOf(
    PropTypes.shape({
      backgroundImage: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      addr1: PropTypes.string.isRequired,
      addr2: PropTypes.string.isRequired,
      city: PropTypes.string.isRequired,
      state: PropTypes.string.isRequired,
      lastInspectionDate: PropTypes.number.isRequired,
      lastInspectionScore: PropTypes.number.isRequired
    }).isRequired
  ).isRequired
};
