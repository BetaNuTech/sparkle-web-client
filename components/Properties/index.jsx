import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { calculateTeamValues } from '../../utils/calculateTeamValues';
import {
  activePropertiesSortFilter,
  sortProperties
} from '../../utils/propertiesSorting';
import {
  fetchDataOfProperties,
  setDataOfProperties
} from '../../redux/ducks/properties/actionCreators';
import {
  selectActiveSortOfProperties,
  selectItemsOfProperties
} from '../../redux/ducks/properties/selectors';
import { fetchDataOfTeams } from '../../redux/ducks/teams/actionCreators';
import { selectItemsOfTeams } from '../../redux/ducks/teams/selectors';
import styles from './Properties.module.scss';
import { Header } from './Header';
import { Sidebar } from './Sidebar';
import { ProfileList } from './ProfileList';
import { MobileProperties } from './MobileProperties';

export const Properties = () => {
  const dispatch = useDispatch();
  const [teamCalculatedValues, setTeamCalculatedValues] = useState([]);
  const activeSort = useSelector(selectActiveSortOfProperties);
  const properties = useSelector(selectItemsOfProperties);
  const teams = useSelector(selectItemsOfTeams);

  // We fetch data only if store with properties is empty.
  // When we switch the page with full store, it doesn't fetch.
  useEffect(() => {
    dispatch(fetchDataOfTeams());

    if (properties.length === 0) {
      dispatch(fetchDataOfProperties());
    }
  }, []);

  // We recalculate properties when properties or teams changes.
  useEffect(() => {
    setTeamCalculatedValues(calculateTeamValues(teams, properties));
  }, [properties, teams]);

  // We sort  properties when activeSort changes.
  useEffect(() => {
    dispatch(
      setDataOfProperties(
        [...properties].sort(
          sortProperties(activeSort.sortBy, activeSort.orderBy)
        )
      )
    );
  }, [activeSort]);

  return (
    <>
      <div className={styles.properties__sortInfoLine}>
        {`Sorted by ${activePropertiesSortFilter(activeSort.sortBy)}`}
      </div>

      <div className={styles.properties__wrapper}>
        <header>
          <Header activeSort={activeSort} properties={properties} />
        </header>

        <div className={styles.properties__main}>
          <ProfileList activeSort={activeSort} properties={properties} />
        </div>

        <aside>
          <Sidebar teams={teams} teamCalculatedValues={teamCalculatedValues} />
        </aside>
      </div>

      <div className={styles.properties__mobile}>
        <MobileProperties
          properties={properties}
          teams={teams}
          teamCalculatedValues={teamCalculatedValues}
        />
      </div>
    </>
  );
};
