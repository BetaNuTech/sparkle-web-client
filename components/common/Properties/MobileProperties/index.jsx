/* eslint-disable max-len */
import styles from './MobileProperties.module.scss';
import { TeamItem } from './TeamItem';
import { PropertyItem } from './PropertyItem';

export const MobileProperties = () => (
  <ul className={styles.mobileProperties}>
    <li className={styles.mobileProperties__item}>
      <header>teams</header>
      <TeamItem name="Team one" />
      <TeamItem name="Team two" />
    </li>

    <li className={styles.mobileProperties__item}>
      <header>properties</header>
      <PropertyItem
        name="Walnut Ridge"
        addr1="1900 Walnut Street"
        mailingAddr2="Bastrop, TX 78602"
      />
      <PropertyItem
        backgroundImage="https://firebasestorage.googleapis.com/v0/b/sparkle-jwc.appspot.com/o/propertyImages%2FdvSsHLv8cxAvIMKv9Gk0.jpeg?alt=media&token=f226942e-bc2e-476f-89dd-d2eee28cd2e4"
        name="1400 Chestnut"
        addr1="1400 Chestnut Street"
        mailingAddr2="Chattanooga, TN 37402"
        lastInspectionEntries="1 Entry [ Last: 100%, May 13 ]"
      />
    </li>
  </ul>
);
