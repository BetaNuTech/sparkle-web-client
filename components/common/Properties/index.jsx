import React, { useState } from 'react';
import styles from './Properties.module.scss';
import { Header } from './Header';
import { ProfileList } from './ProfileList';
import { Sidebar } from './Sidebar';
import { MobileProperties } from './MobileProperties';

export const Properties = () => {
  const [isAscendingSort, setIsAscendingSort] = useState(true);

  return (
    <div className={styles.properties}>
      <div className={styles.properties__wrapper}>
        <header>
          <Header
            isAscendingSort={isAscendingSort}
            setIsAscendingSort={setIsAscendingSort}
          />
        </header>

        <div className={styles.properties__main}>
          <ProfileList isAscendingSort={isAscendingSort} />
        </div>

        <aside>
          <Sidebar />
        </aside>
      </div>

      <div className={styles.properties__mobile}>
        <MobileProperties />
      </div>
    </div>
  );
};
