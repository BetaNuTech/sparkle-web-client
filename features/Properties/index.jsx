import PropTypes from 'prop-types';
import { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { calculateTeamValues } from '../../common/utils/calculateTeamValues';
import {
  sorts,
  activePropertiesSortFilter,
  sortProperties
} from '../../common/utils/propertiesSorting';
import {
  fetchDataOfProperties,
  setDataOfProperties,
  setActiveSortOfProperties
} from '../../app/ducks/properties/actionCreators';
import {
  selectActiveSortOfProperties,
  selectItemsOfProperties
} from '../../app/ducks/properties/selectors';
import { fetchDataOfTeams } from '../../app/ducks/teams/actionCreators';
import { selectItemsOfTeams } from '../../app/ducks/teams/selectors';
import styles from './Properties.module.scss';
import { Header } from './Header';
import { Sidebar } from './Sidebar';
import { ProfileList } from './ProfileList';
import { MobileHeader } from './MobileHeader';
import { MobileProperties } from './MobileProperties';

export const Properties = ({
  isOnline,
  isStaging,
  isNavOpen,
  toggleNavOpen
}) => {
  const dispatch = useDispatch();
  const [teamCalculatedValues, setTeamCalculatedValues] = useState([]);
  const activeSort = useSelector(selectActiveSortOfProperties);
  const activeSortDir = { ...activeSort };
  const properties = useSelector(selectItemsOfProperties);
  const teams = useSelector(selectItemsOfTeams);

  // Fetch data only if store with properties is empty.
  // When we switch the page with full store, it doesn't fetch.
  useEffect(() => {
    dispatch(fetchDataOfTeams());

    if (properties.length === 0) {
      dispatch(fetchDataOfProperties());
    }
  }, []);

  // Recalculate properties when properties or teams changes.
  useEffect(() => {
    setTeamCalculatedValues(calculateTeamValues(teams, properties));
  }, [properties, teams]);

  // Sort properties active sort and direction
  useEffect(() => {
    dispatch(
      setDataOfProperties(
        [...properties].sort(
          sortProperties(activeSort.sortBy, activeSort.orderBy)
        )
      )
    );
  }, [activeSort]);

  // Loop through property
  // sorting options
  const nextPropertiesSort = () => {
    const activeFilter =
      sorts[sorts.indexOf(activeSort.sortBy) + 1] || sorts[0]; // Get next or first

    // Update Property Filter
    dispatch(
      setActiveSortOfProperties({
        ...activeSort,
        sortBy: activeFilter
      })
    );
  };

  // Set sort attribute & direction
  const onSortChange = (key) => (event) => {
    const {
      target: { value }
    } = event;

    if (key === 'orderBy') {
      // Update orderBy filter
      if (activeSortDir[key] === 'asc') activeSortDir[key] = 'desc';
      else if (activeSortDir[key] === 'desc') activeSortDir[key] = 'asc';
    } else {
      // Update sortBy filter
      activeSortDir[key] = value;
    }

    // Update Property Filter
    dispatch(setActiveSortOfProperties(activeSortDir));
  };

  return (
    <>
      <MobileHeader
        title="Properties"
        isNavOpen={isNavOpen}
        toggleNavOpen={toggleNavOpen}
        nextPropertiesSort={nextPropertiesSort}
        isOnline={isOnline}
        isStaging={isStaging}
      />

      <div className={styles.properties__sortInfoLine}>
        {`Sorted by ${activePropertiesSortFilter(activeSort.sortBy)}`}
      </div>

      <div className={styles.properties__wrapper}>
        <Header activeSort={activeSort} onSortChange={onSortChange} />

        <div className={styles.properties__main}>
          <ProfileList activeSort={activeSort} properties={properties} />
        </div>

        <aside>
          <Sidebar teams={teams} teamCalculatedValues={teamCalculatedValues} />
        </aside>
      </div>

      <div className={styles.properties__mobile}>
        <MobileProperties
          properties={properties}
          teams={teams}
          teamCalculatedValues={teamCalculatedValues}
        />
      </div>
    </>
  );
};

Properties.propTypes = {
  isOnline: PropTypes.bool,
  isStaging: PropTypes.bool,
  isNavOpen: PropTypes.bool,
  toggleNavOpen: PropTypes.func
};

Properties.defaultProps = {
  isOnline: false,
  isStaging: false,
  isNavOpen: false,
  toggleNavOpen: () => {}
};
