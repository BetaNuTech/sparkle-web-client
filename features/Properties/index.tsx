import { FunctionComponent, useState, useEffect } from 'react';
import { useMediaQuery } from 'react-responsive';
import { useFirestore } from 'reactfire';
import calculateTeamValues from './utils/calculateTeamValues';
import {
  sorts,
  activePropertiesSortFilter,
  sortProperties
} from './utils/propertiesSorting';
import { useSortBy, useSortDir } from './hooks/sorting';
import useTeams from './hooks/useTeams';
import useProperties from './hooks/useProperties';
import useDeleteProperty from '../../common/hooks/useDeleteProperty';
import useDeleteTeam from './hooks/useDeleteTeam';
import {
  canCreateTeam,
  canCreateProperty
} from '../../common/utils/userPermissions';
import useNotifications from '../../common/hooks/useNotifications'; // eslint-disable-line
import userModel from '../../common/models/user';
import propertyModel from '../../common/models/property';
import teamModel from '../../common/models/team';
import notifications from '../../common/services/notifications';
import globalEvents from '../../common/utils/globalEvents';
import breakpoints from '../../config/breakpoints';
import styles from './styles.module.scss';
import Header from './Header';
import Sidebar from './Sidebar';
import ProfileList from './ProfileList';
import MobileLayout from './MobileLayout';
import DeleteTeamPrompt from './DeleteTeamPrompt';

interface PropertiesModel {
  user: userModel;
  isOnline?: boolean;
  isStaging?: boolean;
  isNavOpen?: boolean;
  toggleNavOpen?(): void;
  forceVisible?: boolean;
}

const Properties: FunctionComponent<PropertiesModel> = ({
  user,
  isOnline,
  isStaging,
  toggleNavOpen,
  forceVisible
}) => {
  const firestore = useFirestore();
  const [teamCalculatedValues, setTeamCalculatedValues] = useState([]);
  const [sortBy, setSortBy] = useSortBy();
  const [sortDir, setSortDir] = useSortDir();
  const { data: properties, memo: propertiesMemo } = useProperties(
    firestore,
    user
  );
  const {
    status: teamsStatus,
    data: teams,
    memo: teamsMemo
  } = useTeams(firestore, user);
  const [sortedProperties, setSortedProperties] = useState([]);

  // Lookup user permissions
  const hasCreateTeamPermission = canCreateTeam(user);
  const hasCreatePropertyPermission = canCreateProperty(user);

  // User notifications setup
  /* eslint-disable */
  const sendNotification = notifications.createPublisher(useNotifications());
  /* eslint-enable */

  // Queue and Delete Property
  const [isDeletePropertyPromptVisible, setDeletePropertyPromptVisible] =
    useState(false);
  const { queuePropertyForDelete, confirmPropertyDelete } = useDeleteProperty(
    firestore,
    sendNotification,
    user
  );
  const openPropertyDeletePrompt = (property: propertyModel) => {
    queuePropertyForDelete(property);
    setDeletePropertyPromptVisible(true);
  };
  const closeDeletePropertyPrompt = () => {
    setDeletePropertyPromptVisible(false);
    queuePropertyForDelete(null);
  };

  // Queue and Delete Team
  const [isDeleteTeamPromptVisible, setDeleteTeamPromptVisible] =
    useState(false);
  const { queueTeamForDelete, confirmTeamDelete } = useDeleteTeam(
    firestore,
    sendNotification,
    user
  );
  const openTeamDeletePrompt = (team: teamModel) => {
    queueTeamForDelete(team);
    setDeleteTeamPromptVisible(true);
  };
  const closeDeleteTeamPrompt = () => {
    setDeleteTeamPromptVisible(false);
    queueTeamForDelete(null);
  };

  // Responsive queries
  const isMobileorTablet = useMediaQuery({
    maxWidth: breakpoints.tablet.maxWidth
  });
  const isDesktop = useMediaQuery({
    minWidth: breakpoints.desktop.minWidth
  });

  // Apply properties sort order
  const applyPropertiesSort = () =>
    [...properties].sort(sortProperties(sortBy, sortDir));

  // Update team calculated values
  useEffect(() => {
    function recalcTeamComputedValues() {
      if (teamsStatus === 'success') {
        setTeamCalculatedValues(calculateTeamValues(teams, properties));
      }
    }

    recalcTeamComputedValues();
  }, [teamsStatus, teamsMemo, propertiesMemo]); // eslint-disable-line

  useEffect(() => {
    function resortProperties() {
      setSortedProperties(applyPropertiesSort());
    }

    resortProperties();
  }, [propertiesMemo, sortBy, sortDir]); // eslint-disable-line

  // Loop through property
  // sorting options
  const nextPropertiesSort = () => {
    const activeSortValue = sorts[sorts.indexOf(sortBy) + 1] || sorts[0]; // Get next or first
    const descOrderKeys = ['lastInspectionDate', 'lastInspectionScore'];
    const isDescOrderValue = descOrderKeys.includes(activeSortValue);

    // Check if sort direction is asc then
    // change to desc if it is either lastInspectionDate or lastInspectionScore
    if (isDescOrderValue && sortDir === 'asc') {
      setSortDir('desc');
    } else if (!isDescOrderValue && sortDir === 'desc') {
      // If sort desc and it is other than values
      // in `descOrderKeys` then change to asc
      setSortDir('asc');
    }
    // Update Property sort
    setSortBy(activeSortValue);
    globalEvents.trigger('visibilityForceCheck');
  };

  // Set sort attribute & direction
  const onSortChange = (key: string) => (evt: { target: HTMLInputElement }) => {
    const {
      target: { value }
    } = evt;

    // Update sort direction
    if (key === 'sortDir') {
      setSortDir(sortDir === 'asc' ? 'desc' : 'asc');
    } else {
      setSortBy(value); // Update sort by
    }

    globalEvents.trigger('visibilityForceCheck');
  };

  return (
    <>
      {isMobileorTablet && (
        <MobileLayout
          properties={sortedProperties}
          teams={teams}
          teamCalculatedValues={teamCalculatedValues}
          isDeletePropertyPromptVisible={isDeletePropertyPromptVisible}
          confirmPropertyDelete={confirmPropertyDelete}
          openPropertyDeletePrompt={openPropertyDeletePrompt}
          closeDeletePropertyPrompt={closeDeletePropertyPrompt}
          openTeamDeletePrompt={openTeamDeletePrompt}
          isOnline={isOnline}
          isStaging={isStaging}
          canAddTeam={hasCreateTeamPermission}
          canAddProperty={hasCreatePropertyPermission}
          toggleNavOpen={toggleNavOpen}
          nextPropertiesSort={nextPropertiesSort}
          sortBy={sortBy}
          activePropertiesSortFilter={activePropertiesSortFilter}
          forceVisible={forceVisible}
        />
      )}

      {/* Desktop Header & Content */}
      {isDesktop && (
        <div className={styles.properties__container}>
          <Header
            sortBy={sortBy}
            sortDir={sortDir}
            onSortChange={onSortChange}
            canAddTeam={hasCreateTeamPermission}
            canAddProperty={hasCreatePropertyPermission}
          />

          <div className={styles.properties__main}>
            <ProfileList
              properties={sortedProperties}
              forceVisible={forceVisible}
            />
          </div>

          <aside>
            <Sidebar
              teams={teams}
              openTeamDeletePrompt={openTeamDeletePrompt}
              teamCalculatedValues={teamCalculatedValues}
            />
          </aside>
        </div>
      )}

      <DeleteTeamPrompt
        isVisible={isDeleteTeamPromptVisible}
        onClose={closeDeleteTeamPrompt}
        onConfirm={confirmTeamDelete}
      />
    </>
  );
};

Properties.defaultProps = {
  isOnline: false,
  isStaging: false,
  isNavOpen: false,
  toggleNavOpen: () => {}, // eslint-disable-line
  forceVisible: false
};

export default Properties;
