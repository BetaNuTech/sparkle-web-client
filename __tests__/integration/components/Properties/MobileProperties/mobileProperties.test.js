import React from 'react';
import { shallow } from 'enzyme';
import propertiesMock from '../../../../../__mocks__/propertiesMock.json';
import { MobileProperties } from '../../../../../components/Properties/MobileProperties';

describe('MobileProperties component', () => {
  describe('rendering', () => {
    let wrapper;

    beforeEach(() => {
      const props = {
        properties: propertiesMock
      };
      wrapper = shallow(<MobileProperties {...props} />);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
