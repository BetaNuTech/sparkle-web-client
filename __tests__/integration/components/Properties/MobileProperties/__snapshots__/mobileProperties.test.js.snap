// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileProperties component rendering renders correctly 1`] = `ShallowWrapper {}`;
